### PR TITLE
refactor(SettingsAccountStyle): code

### LINKED ISSUE #387: settings-account-style

### PR DESCRIPTION
**Pull request by samandarkuchkarov**
Fixes #387 

## Proposed Changes

-Change the imports to absolute:
-Move text to i18n
-Change the colors from the constants to similar ones in Color
-Transferring the screen logic to components
-Replace navigation hooks with typed navigation hooks in folder src/hooks

## Checklist

- [x] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

**Review by gHashTag — APPROVED** [review]

**Review by vivalaakam — CHANGES_REQUESTED** [review]
I think that we can hide most of functions from `SettingsAccountStyleScreen` inside `SettingsAccountStyle` and pass there wallet and onPressApply

File: src/i18n.ts [hunk 1]
```diff
@@ -189,6 +189,14 @@ export enum I18N {
   backupFinishCongratulation,
   backupFinishSuccess,
   backupFinishFinish,
+  settingsAccountStyleChoseColor,
+  settingsAccountStyleGenerate,
+  settingsAccountStyleUseStyle,
+  settingsAccountStyleAlreadyUsed,
+  settingsAccountStyleFlat,
+  settingsAccountStyleGradient,
+  settingsAccountStyleCircle,
```
Comment (gHashTag) [inline src/i18n.ts:198]: put the translations in alphabetical order of 1 letter

**Review by gHashTag — CHANGES_REQUESTED** [review]
Fix comment

File: src/components/settings-account-style.tsx [hunk 2]
```diff
@@ -50,32 +53,35 @@ const patternVariants = [
...
-  isStyleChanged,
+  setPattern,
+  setColors,
+  setPatternStyle,
+  setCardStyle,
+  wallet,
 }: SettingsAccountStyleProps) => {
   const opacity = useSharedValue(1);
   const heightCard = 192;
+  const timerRef: {current: NodeJS.Timeout | null} = useRef(null);
```
Comment (gHashTag) [inline src/components/settings-account-style.tsx:81]: Try use generics
```js
const timerRef = useRef<NodeJS.Timeout | null>(null);
```

File: src/components/settings-account-style.tsx [hunk 3]
```diff
@@ -110,10 +110,10 @@ export const SettingsAccountStyle = ({
...
     } = wallet;
     if (
-      WCardStyle !== cardStyle ||
-      colors[0] !== colorFrom ||
-      colors[1] !== colorTo ||
-      colors[2] !== colorPattern ||
+      WCardStyle !== cardStyle ??
+      colors[0] !== colorFrom ??
+      colors[1] !== colorTo ??
+      colors[2] !== colorPattern ??
```
Comment (gHashTag) [inline src/components/settings-account-style.tsx:116]: Need to think before replacing `||` with `??`, here you've broken the logic

File: src/screens/settings-account-style.tsx [hunk 4]
```diff
@@ -15,7 +15,7 @@ export const SettingsAccountStyleScreen = () => {
   const [pattern, setPattern] = useState<string>(wallet.pattern);
 
   const [cardStyle, setCardStyle] = useState<WalletCardStyle>(
-    wallet.cardStyle || WalletCardStyle.flat,
+    wallet.cardStyle ?? WalletCardStyle.flat,
```
Comment (gHashTag) [inline src/screens/settings-account-style.tsx:18]: It's probably okay here, if you checked

**Review by gHashTag — CHANGES_REQUESTED** [review]
Fix `??`

File: src/components/settings-account-style.tsx (hunk 3 above)
Comment (samandarkuchkarov) [reply src/components/settings-account-style.tsx]: done

**Comment by gHashTag** [comment]
Fix translation alphabet order

File: src/i18n.ts (hunk 1 above)
Comment (samandarkuchkarov) [reply src/i18n.ts:198]: done

**Comment by samandarkuchkarov** [comment]
change

> Fix `??`

done

**Comment by samandarkuchkarov** [comment]
> I think that we can hide most of functions from `SettingsAccountStyleScreen` inside `SettingsAccountStyle` and pass there wallet and onPressApply

done

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**